deid/data/rsna-anon-2022.dicom

```diff
@@ -0,0 +1,3272 @@
+REMOVE TargetMaterialSeq
+REMOVE SNRThreshold
+REMOVE ImageScaleRepresentation
+REMOVE RefPTOSeq
+REMOVE RefTDRInstanceSeq
+REMOVE PTOLocationDescription
+REMOVE AnomalyLocatorIndicatorSeq
+REMOVE AnomalyLocatorIndicator
+REMOVE PTORegionSeq
+REMOVE InspectionSelectionCriteria
+REMOVE SecondaryInspectionMethodSeq
+REMOVE PRCSToRCSOrientation
+REMOVE MACParametersSeq
+REMOVE SharedFunctionalGroupsSeq
+REMOVE PerFrameFunctionalGroupsSeq
+REMOVE WaveformSeq
+REMOVE ChannelMinimumValue
+REMOVE ChannelMaximumValue
+REMOVE WaveformBitsAllocated
+REMOVE WaveformSampleInterpretation
+REMOVE WaveformPaddingValue
+REMOVE WaveformData
+REMOVE FirstOrderPhaseCorrectionAngle
+REMOVE SpectroscopyData
+REMOVE DigitalSignaturesSeq
+REMOVE DataSetTrailingPadding
+
+
+
+BLANK ClinicalTrialSponsorName
+BLANK ClinicalTrialProtocolID
+BLANK ClinicalTrialProtocolName
+BLANK ClinicalTrialSiteID
+BLANK ClinicalTrialSiteName
+BLANK ClinicalTrialSubjectID
+BLANK ClinicalTrialSubjectReadingID
+BLANK ClinicalTrialTimePointID
+BLANK CoordinatingCenterName
+BLANK ClinicalTrialProtocolEthicsCommitteeName
+BLANK XRaySourceID
+BLANK XRayDetectorID
+BLANK StudyID
+
+# Dates to modify
+REMOVE StudyDate
+REMOVE SeriesDate
+REMOVE AcquisitionDate
+REMOVE ContentDate
+REMOVE OverlayDate
+REMOVE CurveDate
+REMOVE AcquisitionDatetime
+REMOVE RadiopharmaceuticalStartDateTime
+REMOVE RadiopharmaceuticalStopDateTime
+REMOVE SourceStartDateTime
+REMOVE SourceEndDateTime
+REMOVE StartAcquisitionDateTime
+REMOVE EndAcquisitionDateTime
+
+ADD StudyDate func:process_date
+ADD SeriesDate func:process_date
+ADD AcquisitionDate func:process_date
+ADD ContentDate func:process_date
+ADD OverlayDate func:process_date
+ADD CurveDate func:process_date
+ADD AcquisitionDatetime func:process_date
+ADD RadiopharmaceuticalStartDateTime func:process_date
+ADD RadiopharmaceuticalStopDateTime func:process_date
+ADD SourceStartDateTime func:process_date
+ADD SourceEndDateTime func:process_date
+ADD StartAcquisitionDateTime func:process_date
+ADD EndAcquisitionDateTime func:process_date
+
+# UIDs to process
+REMOVE SOPInstanceUID
+REMOVE StudyInstanceUID
+REMOVE SeriesInstanceUID
+REMOVE FrameOfReferenceUID
+REMOVE SynchronizationFrameOfReferenceUID
+REMOVE ConcatenationUID
+REMOVE DimensionOrganizationUID
+REMOVE PaletteColorLUTUID
+REMOVE LargePaletteColorLUTUID
+
+ADD SOPInstanceUID func:process_uid
+ADD StudyInstanceUID func:process_uid
+ADD SeriesInstanceUID func:process_uid
+ADD FrameOfReferenceUID func:process_uid
+ADD SynchronizationFrameOfReferenceUID func:process_uid
+ADD ConcatenationUID func:process_uid
+ADD DimensionOrganizationUID func:process_uid
+ADD PaletteColorLUTUID func:process_uid
+ADD LargePaletteColorLUTUID func:process_uid
+
+## Main identifiers to process
+REMOVE AccessionNumber
+REMOVE PatientName
+REMOVE PatientID
+REMOVE PatientAge
+
+ADD AccessionNumber var:accession_number
+ADD PatientName var:patient_name
+ADD PatientID var:patient_id
+ADD PatientAge var:patient_age
+
```

Comment:
Let's remove empty space at the end.

Reply:
Are you sure about adding `func:process_uid` here? This function won't exist for most users. We do have plan to support deid provided functions (discussion was here) https://github.com/pydicom/deid/pull/203 but no implementation yet.

Reply:
happy to remove the trailing empty lines. I think having the functions make sense provided the tags are removed first, at least with var: a missing lookup meant the tag was not added back to the file, I think that behaviour is ok for these. What do you think?

Reply:
I guess I don't quite understand, is this a difference between `func:` and `var:`? Because deid.dicom includes `var:`'s that are unexpected when working from within python (I haven't used the command line scripts). But I'm happy to comment them out anyway

Reply:
Var and func both won't work without the user providing functions and variables in the lookup. E.g., I would need to define that function or a variable of a particular name and pass it to the function to perform the parsing. We generally want these core provided recipes to not require that extra customization - e.g., most people are going to try it and be surprised that there is an extra function or variable dependency!

However, if / when we have these deid provided functions, we will easily be able to add them to the defaults. Does that make sense? Some of the documentation examples  of using func/var might help.

Reply:
I'm not sure what you mean they won't work - it appears to me that the default behaviour is to leave the tag unmodified, which is why for each `ADD tag var:` and `ADD tag func:` I preceded them with `REMOVE tag`, that way if the function or variable isn't supplied the tag is removed. Quick example:

```python
import os
import tempfile
from pydicom.dataset import FileDataset, FileMetaDataset
from pydicom.uid import UID

# Create some temporary filenames
suffix = '.dcm'
filename_little_endian = tempfile.NamedTemporaryFile(suffix=suffix).name
filename_big_endian = tempfile.NamedTemporaryFile(suffix=suffix).name

print("Setting file meta information...")
# Populate required values for file meta information
file_meta = FileMetaDataset()
file_meta.MediaStorageSOPClassUID = UID('1.2.840.10008.5.1.4.1.1.2')
file_meta.MediaStorageSOPInstanceUID = UID("1.2.3")
file_meta.ImplementationClassUID = UID("1.2.3.4")

print("Setting dataset values...")
# Create the FileDataset instance (initially no data elements, but file_meta
# supplied)
ds = FileDataset(filename_little_endian, {},
                 file_meta=file_meta, preamble=b"\0" * 128)

ds.PatientAge = "206"

from deid.config import DeidRecipe
from deid.dicom.parser import DicomParser
recipe = DeidRecipe()
recipe.deid["header"].append({'action' : 'ADD', 'field' : 'PatientAge', 'value' : 'var:new_age'})
dparser = DicomParser(ds, recipe = recipe)

dparser.parse()
ds
```

Retains the PatientAge of 206 and does not error.

```python
recipe2 = DeidRecipe()
recipe2.deid["header"].append({'action' : 'REMOVE', 'field' : 'PatientAge'})
recipe2.deid["header"].append({'action' : 'ADD', 'field' : 'PatientAge', 'value' : 'var:new_age'})

dparser = DicomParser(ds, recipe = recipe2)
dparser.parse()

ds
```

shows the PatientAge field deleted

and finally

```python
dparser.lookup["new_age"] = 102
dparser.parse()

ds 
```

shows the modified age of 102

But your library your choice! I'm happy to comment those actions out until #207 is complete.

Reply:
Sorry I can be more clear - it will run but it will emit this error message: https://github.com/pydicom/deid/blob/2718b7e5d9aa155575f26656b890767728cd669d/deid/utils/actions.py#L54-L56.

Hmm. So I guess I don't feel very strongly about this, so if your preference is to leave them, that's okay with me. But likely we will need to refactor a bit if/when we add default functions (do you agree we should provide some default to plug in to this recipe?)

Reply:
ahh interesting, I've never seen those warnings when working in the python repl!

Yes I think including defaults would be helpful, autogenerating new UIDs is a particularly appealing example. The RSNA anonymizer hashes the date, we could provide functionality like that as well if it was desired. I was just trying to stick to the spirit of their recipe.